### PR TITLE
ISSUE-3957: Parallel import data error

### LINKED ISSUE #3957: Parallel import data was error
load ontime data with parallel:

```
time ls *.csv|xargs -I{} -P 4 curl -H "insert_sql:insert into ontime1 format CSV" -H "csv_header:1" -F  "upload=@{}"  -XPUT http://localhost:8000/v1/streaming_load
```

{"id":"50dbd689-c838-41a4-9092-f967ef60d0d6","state":"SUCCESS","stats":{"read_rows":0,"read_bytes":0},"error":null}{"id":"a93c46db-4474-416c-87d1-6dbbffa9dc56","state":"SUCCESS","stats":{"read_rows":0,"read_bytes":0},"error":null}{"id":"0e490a29-e88f-42ef-8f37-27ceb452696e","state":"SUCCESS","stats":{"read_rows":0,"read_bytes":0},"error":null}{"id":"c77e9935-a25d-42eb-b812-d1782afb0653","state":"SUCCESS","stats":{"read_rows":0,"read_bytes":0},"error":null}{"id":"2803d342-91b7-4590-ab0c-7e79454b674a","state":"SUCCESS","stats":{"read_rows":0,"read_bytes":0},"error":null}{"id":"6a8bd3a4-c488-4da0-bb0d-5e34bb2ffd8e","state":"SUCCESS","stats":{"read_rows":0,"read_bytes":0},"error":null}{"id":"854f332d-e34a-472e-8524-586ac0dc575e","state":"SUCCESS","stats":{"read_rows":0,"read_bytes":0},"error":null}{"id":"cc28b528-6012-4ad5-b52d-5fa533c00658","state":"SUCCESS","stats":{"read_rows":0,"read_bytes":0},"error":null}{"id":"eb302b59-dfcc-44fa-8ae5-a06fbd032376","state":"SUCCESS","stats":{"read_rows":0,"read_bytes":0},"error":null}Code: 2009, displayText = targeting version Exact(595), current version 604.Code: 2009, displayText = targeting version Exact(595), current version 604.Code: 2009, displayText = targeting version Exact(595), current version 604.{"id":"ad3b3fa0-d948-4f69-a4d6-fe816693b019","state":"SUCCESS","stats":{"read_rows":441670,"read_bytes":318949668},"error":null}{"id":"3edb2f98-c7da-4a76-afe2-172f33fabf26","state":"SUCCESS","stats":{"read_rows":412579,"read_bytes":297871635},"error":null}Code: 2009, displayText = targeting version Exact(604), current version 606.Code: 2009, displayText = targeting version Exact(604), current version 606.Code: 2009, displayText = targeting version Exact(604), current version 606.{"id":"272d9f16-b400-4187-b58c-2335e3db8d39","state":"SUCCESS","stats":{"read_rows":427325,"read_bytes":308543754},"error":null}Code: 2009, displayText = targeting version Exact(606), current version 607.Code: 2009, displayText = targeting version Exact(606), current version 607.Code: 2009, displayText = targeting version Exact(606), current version 607.{"id":"210c3c74-7a21-4183-8cec-dc225c652145","state":"SUCCESS","stats":{"read_rows":441118,"read_bytes":318477307},"error":null}Code: 2009, displayText = targeting version Exact(607), current version 608.Code: 2009, displayText = targeting version Exact(607), current version 608.Code: 2009, displayText = targeting version Exact(607), current version 608.{"id":"c18b7b92-e5f9-4427-8901-35b4cba7cf92","state":"SUCCESS","stats":{"read_rows":395176,"read_bytes":285378317},"error":null}Code: 2009, displayText = targeting version Exact(608), current version 609.Code: 2009, displayText = targeting version Exact(608), current version 609.Code: 2009, displayText = targeting version Exact(608), current version 609.{"id":"f1aa1aab-f6da-4ce0-98bc-5c8a6d49e60b","state":"SUCCESS","stats":{"read_rows":405604,"read_bytes":292980284},"error":null}Code: 2009, displayText = targeting version Exact(609), current version 610.Code: 2009, displayText = targeting version Exact(609), current version 610.Code: 2009, displayText = targeting version Exact(609), current version 610.{"id":"8de1df6c-4c2c-4145-89a6-a1ed45842b51","state":"SUCCESS","stats":{"read_rows":417166,"read_bytes":301251519},"error":null}Code: 2009, displayText = targeting version Exact(610), current version 611.Code: 2009, displayText = targeting version Exact(610), current version 611.Code: 2009, displayText = targeting version Exact(610), current version 611.{"id":"b8c81e99-c08c-4990-b599-f70c233d3f74","state":"SUCCESS","stats":{"read_rows":428781,"read_bytes":309749140},"error":null}Code: 2009, displayText = targeting version Exact(611), current version 612.Code: 2009, displayText = targeting version Exact(611), current version 612.
Code: 2009, displayText = targeting version Exact(611), current version 612.{"id":"ef55c350-77e2-4d4c-9967-379424a04380","state":"SUCCESS","stats":{"read_rows":444320,"read_bytes":320929005},"error":null}Code: 2009, displayText = targeting version Exact(612), current version 613.Code: 2009, displayText = targeting version Exact(612), current version 613.Code: 2009, displayText = targeting version Exact(612), current version 613.{"id":"2cd49104-f83d-4975-84c7-83b1eb9cf90b","state":"SUCCESS","stats":{"read_rows":455057,"read_bytes":328721797},"error":null}Code: 2009, displayText = targeting version Exact(613), current version 614.Code: 2009, displayText = targeting version Exact(613), current version 614.Code: 2009, displayText = targeting version Exact(613), current version 614

### PR DESCRIPTION
**Pull request by dantengsky**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Fixing:
- issue #3957

## Changelog

- Improvement
- Not for changelog (changelog entry is not required)

## Related Issues

Fixes #3957

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

File: query/src/storages/fuse/operations/commit.rs [hunk 1]
```diff
@@ -109,12 +158,12 @@ impl FuseTable {
     }
 
     async fn commit_to_meta_server(
-        &self,
+        tbl_id: &TableIdent,
```
Comment (BohuTANG) [inline query/src/storages/fuse/operations/commit.rs:161]: How about re-order the arguments to:
```
  ctx: &QueryContext,
  tbl_id: &TableIdent,
```
?

Comment (dantengsky) [reply query/src/storages/fuse/operations/commit.rs:161]: :ok_hand:

File: query/src/storages/fuse/operations/commit.rs [hunk 2]
```diff
@@ -33,18 +36,65 @@ use crate::storages::fuse::operations::TableOperationLog;
...
+        let tid = self.table_info.ident.table_id;
+        let mut tbl = self;
+        let mut latest: Arc<dyn Table>;
+        loop {
+            match tbl
+                .try_commit(ctx.as_ref(), &operation_log, overwrite)
+                .await
+            {
+                Ok(_) => break Ok(()),
+                Err(e) if e.code() == ErrorCode::table_version_mismatched_code() => {
```
Comment (BohuTANG) [inline query/src/storages/fuse/operations/commit.rs:59]: It would be better to add a `tracing::error!` to log the re-try error here

Comment (dantengsky) [reply query/src/storages/fuse/operations/commit.rs:59]: ok

**Comment by PsiACE** [comment]
cc @dantengsky , ref https://github.com/datafuselabs/databend/pull/3981/commits/16799a6f0b1a1a66f357cc69064bf28167d8a6bb

**Comment by dantengsky** [comment]
@PsiACE got it, thanks!

File: query/src/storages/fuse/operations/commit.rs [hunk 3]
```diff
@@ -33,18 +39,107 @@ use crate::storages::fuse::operations::TableOperationLog;
...
-        let prev = self.read_table_snapshot(ctx.as_ref()).await?;
+                            let catalog = ctx.get_catalog();
+                            let (ident, meta) = catalog.get_table_meta_by_id(tid).await?;
+                            let table_info: TableInfo = TableInfo {
+                                ident,
+                                desc: "".to_owned(),
+                                name,
+                                meta: meta.as_ref().clone(),
+                            };
+                            latest = catalog.get_table_by_info(&table_info)?;
```
Comment (Veeupup) [inline query/src/storages/fuse/operations/commit.rs:109]: table version is increased in `metasrv`, so here we can only use the latest table info?

**Review by PsiACE — APPROVED** [review]

File: query/src/storages/fuse/operations/commit.rs (hunk 3 above)
Comment (dantengsky) [reply query/src/storages/fuse/operations/commit.rs:109]: exactly

**Pull request merged**